pandas/tests/groupby/test_transform.py

```diff
@@ -880,3 +880,14 @@ def test_transform_absent_categories(func):
     result = getattr(df.y.groupby(df.x), func)()
     expected = df.y
     assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize('func', ['ffill', 'bfill'])
+@pytest.mark.parametrize('key, val', [('level', 0), ('by', Series([0]))])
+def test_ffill_not_in_axis(func, key, val):
+    # GH 21521
+    df = pd.DataFrame([[0]])
```

Comment:
Can you make this test a little stronger? Something that actually fills would be preferable

Reply:
So the groupby ffill methods already have some stronger tests, `test_group_fill_methods` in the same file. This test is specifically there to check for bug #21521, which I think it's sufficient for?